folio-org/mod-circulation
# CIRC-527: Use batch request update endpoint to update request positions

https://issues.folio.org/browse/CIRC-527: Use Request Batch API when modifying item request queue

## Purpose
The concurrent update of request positions is replaced now with single batch update request execution.

## Testing
Following cases have been verified manually:
* Reorder requests;
* Delete request;
* Move request;
* Cancel a request;
* Create a request;
* Check out an item with fulfilling request at the top.

Test coverage of `org.folio.circulation.domain.UpdateRequestQueue` (this is the only consumer of the position update method) is 100%:

![image](https://user-images.githubusercontent.com/9094650/68391022-4773f200-016f-11ea-971c-d5cf04112fe7.png)

## Learning


## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.
